### PR TITLE
ENG-1293: Update `rasa-calm-demo` to version 3.10.1

### PR DESCRIPTION
**Pull request by AsadHasan-Rasa**
## Description
Update rasa calm demo to version 3.10.1. As part of that:
1. Cherry-picked commits from open PRs #47 and #48.
a.     Updated/fixed effected tests accordingly.
b.     Added tests to verify that link to `pattern_human_handoff` is triggered in relevant conditions (when 2 clarifications reached or when 3 auth fails during portfolio-check)
2. Updated README, and config files as per 3.10 changes.
3. (side-change) Also running tests in CI with `--e2e-results` flag and saving test results for unsuccessful CI steps, to ease with local troubleshooting of test failures.

Once above changes are merged to `release-3.10`, then `release-3.10` can be merged to `main` to update to `3.10.1`.

**Note**: ~~Currently CI pipeline is blocked though due to OpenAI rate-limit issue: `OpenAIException - Error code: 429 - ***'error': ***'message': 'You exceeded your current quota, please check your plan and billing details.` (issue raised in [dev-tribe](https://rasa-hq.slack.com/archives/C061J0LGHU0/p1726245691626949)), so currently keeping this PR as Draft.~~  Sorted: OpenAI usage budget limit is now updated/increased and this issue is resolved, so CI pipeline was able to proceed and all tests passing now ✅ 

## TODOs
- [x] compared flaky tests with the [known list of flaky tests steps](https://www.notion.so/rasa/Flaky-E2E-Test-Steps-63864d3d8c7b4427a0f3df8052e39f21)

### CODE REVIEW

File: data/flows/patterns.yml [hunk 1]
```diff
@@ -51,4 +51,31 @@ flows:
...
+      - action: action_clarify_flows
+        next:
+          - if: context.names contains "add a contact"
+            then:
+              - link: add_contact
+          - else: clarify_options
+      - id: clarify_options
+        action: utter_clarification_options_rasa
+
+  pattern_clarification_with_human_handoff:
```
Comment (ancalita) [inline data/flows/patterns.yml:69]: Is this a new pattern added to the default file in rasa-private? Or should this be overriding pattern handoff?

File: README.md [hunk 2]
```diff
@@ -444,4 +444,39 @@ make rasa-test-multistep
 or
 ```commandline
 run rasa test e2e e2e_tests/multistep
-```
\ No newline at end of file
+```
+
+#### E2E tests with assertions
```
Comment (ancalita) [inline README.md:449]: Thanks for updating the readme 👍🏻

**Review by ancalita — APPROVED** [review]
The changes in the 2 e2e tests with assertions that were previous failing LGTM 👍🏻 I didn't look in depth at other changes, apart from leaving a question a new pattern.

File: data/flows/patterns.yml (hunk 1 above)
Comment (AsadHasan-Rasa) [reply data/flows/patterns.yml:69]: iiuc this is a new pattern added to `rasa-private` recently in #966

Comment (ancalita) [reply data/flows/patterns.yml:69]: This file on `main` doesn't contain such pattern, so Rasa will not classify it as a pattern, but as a user defined flow: https://github.com/RasaHQ/rasa-private/blob/main/rasa/dialogue_understanding/patterns/default_flows_for_patterns.yml

Comment (ancalita) [reply data/flows/patterns.yml:69]: I looked at the PR and the changelog mentions that `pattern_human_handoff` can now be linked in other flows, therefore this flow should be overriding that pattern name.

**Comment by AsadHasan-Rasa** [comment]
Currently CI blocked again due to: https://rasa-hq.slack.com/archives/C36SS4N8M/p1726502977324429

**Comment by AsadHasan-Rasa** [comment]
> Currently CI blocked again due to: https://rasa-hq.slack.com/archives/C36SS4N8M/p1726502977324429

Issue seems resolved now and CI pipeline (and all tests therein) passed ✅ https://github.com/RasaHQ/rasa-calm-demo/actions/runs/10886834156

**Pull request merged**